src/Sdk/Handlers/RequestHandler.php

```diff
@@ -102,10 +102,17 @@ public function executeAndRespond(
             return null;
         }
 
-        $type = \mb_strtolower($action) === self::LIST ?
-            \sprintf('%s[]', \get_class($entity)) : \get_class($entity);
+        // If the response content is blank, return an empty array
+        $content = $response->getContent();
+        if (\trim($content) === '') {
+            return [];
```

Comment:
This isnt correct. You need to pass an appropriate empty value to deserialise as content because the executeAndRespond method is expected to return objects of specific types.

Since we're only ever parsing json here, $content should become `'[]'` when `content === ''`

Reply:
should the content not become `'{}'` instead?